### PR TITLE
feat(generator): add `rails generate pgbus:update` to convert YAML to Ruby DSL

### PR DESCRIPTION
**Pull request by mhenrixon**
## Summary

PR 6 of 10. Adds a Rails generator that reads an existing `config/pgbus.yml` and produces `config/initializers/pgbus.rb` using all the niceties from PRs 1-5: capsule string DSL, ActiveSupport::Duration coercion, auto-tuned pool size, named capsules.

```bash
bin/rails generate pgbus:update
```

The original YAML file is left in place. The generator tells the user to delete it once they've verified the new initializer works.

## Architecture

- **`Pgbus::Generators::ConfigConverter`** (`lib/pgbus/generators/`) — pure logic, no Rails generator framework dependency. Does the YAML → Ruby translation. 19 unit tests.
- **`Pgbus::Generators::UpdateGenerator`** (`lib/generators/pgbus/`) — thin `Rails::Generators::Base` wrapper that reads/writes files. 5 tests including end-to-end.

## Conversion rules

1. **Settings matching the gem default are dropped.** No point restating them — the initializer should only show user intent.
2. **Deprecated settings are dropped silently.** Today only `pool_size` (auto-tuned now). PRs 7-8 will add more.
3. **Duration-shaped values become `Duration` literals**: `604800` → `7.days`, `600` → `10.minutes`, `30` → `30.seconds`.
4. **`workers` array becomes the string DSL** when all capsules are simple (just queues + threads). Capsules with extras (`single_active_consumer`, `consumer_priority`, `prefetch_limit`) fall back to `c.capsule :name, ...` form.
5. **Settings that vary across environments** are wrapped:
   - 2 envs, dev with nil → `unless Rails.env.development?`
   - Otherwise → `case Rails.env when ... end`

## Example transformation

Input (multi-environment YAML):

```yaml
default: &default
  visibility_timeout: 600
  workers:
    - queues: ["*"]
      threads: 5
  max_jobs_per_worker: 10000
  max_memory_mb: 512

development:
  <<: *default
  max_jobs_per_worker: ~
  max_memory_mb: ~

production:
  <<: *default
  workers:
    - queues: ["*"]
      threads: 15
  max_memory_mb: 4096
```

Generated output:

```ruby
# frozen_string_literal: true
#
# Generated by `rails generate pgbus:update` from config/pgbus.yml.
# Review and adjust as needed, then delete config/pgbus.yml.

Pgbus.configure do |c|
  c.workers = case Rails.env
    when "development" then "*: 5"
    when "production" then "*: 15"
  end
  c.visibility_timeout = 10.minutes
  c.max_jobs_per_worker = 10_000 unless Rails.env.development?
  c.max_memory_mb = 4_096 unless Rails.env.development?
end
```

What got dropped:
- `pool_size`, `pool_timeout`, `notify_throttle_ms`, `listen_notify`, `max_retries`, `idempotency_ttl`, `archive_retention`, `stats_enabled`, `stats_retention` (all match defaults)
- `queue_prefix`, `default_queue` (match defaults)

What got kept:
- `workers` (different per env → case block)
- `visibility_timeout` (formatted as `10.minutes`)
- `max_jobs_per_worker` and `max_memory_mb` (with `unless Rails.env.development?` guards)

## Test plan

- [x] 19 unit tests on `ConfigConverter` covering: minimal config, multi-queue capsules, capsules with extras, duration formatting, non-divisible seconds, multi-environment overrides, 3+ env case blocks, deprecated setting dropping, install template default detection, frozen string literal header, generated-from comment, file-based YAML loading, missing-file errors
- [x] 5 generator tests covering: class metadata, `--source`/`--destination` options, end-to-end YAML→initializer conversion via the actual generator class
- [x] 923 unit tests pass (789 + 134 from PRs 1-5 + 34 from this PR), 0 failures
- [x] 97 integration tests pass, 0 failures
- [x] Rubocop clean

## Coming next

- **PR 7-8**: Cull constants. Move silent settings (`pool_timeout`, `polling_interval`, `notify_throttle_ms`, `dispatch_interval`, archive compaction tuning, circuit breaker tuning, web tuning, etc.) out of `Configuration` and into constants in their owning classes. The converter will then automatically drop them too — once they're not in `KNOWN_SETTINGS`, the converter doesn't know about them.
- **PR 9**: README reorg
- **PR 10**: Drop YAML loader (next minor version)

## Summary by CodeRabbit

* **New Features**
  * Added a generator to migrate a YAML configuration into a Ruby initializer with configurable source/destination, source validation, and post-install guidance.

* **New Features**
  * Added a converter that emits a cleaned, environment-aware initializer, normalizes defaults, omits deprecated settings, and formats durations and numbers for readability.

* **Tests**
  * Added comprehensive specs covering end-to-end generator behavior and converter output/edge cases.

### CODE REVIEW

File: lib/generators/pgbus/update_generator.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        destination_path = File.expand_path(options[:destination], destination_root)
+
+        unless File.exist?(source_path)
+          say_status :error, "#{options[:source]} not found", :red
+          exit 1
+        end
+
+        ruby_source = ConfigConverter.from_yaml(source_path)
+        create_file destination_path, ruby_source
+      end
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply lib/generators/pgbus/update_generator.rb:44]: Fixed in 9a9638a. Replaced `exit 1` with `raise Thor::Error, "Source file not found: ..."` — Thor catches it, prints in red, exits cleanly without a Ruby backtrace, and is testable from RSpec.

File: spec/pgbus/generators/config_converter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      end
+
+      it "drops pool_size (auto-tuned now)" do
+        expect(output).not_to include("pool_size")
+      end
+
+      it "still emits the workers (pool_size is what was dropped)" do
+        expect(output).to include('c.workers = "*: 15"')
+      end
+    end
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply spec/pgbus/generators/config_converter_spec.rb:175]: Fixed in 9a9638a. Split into three focused contexts: (1) "with deprecated settings" containing only `pool_size`, (2) "with non-default pool_timeout" asserting emission with a value of 10, (3) "with pool_timeout matching the default" asserting it gets dropped. The original test passed for the wrong reason on pool_timeout — it would only have caught a real regression by coincidence.

File: lib/generators/pgbus/update_generator.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        destination_path = File.expand_path(options[:destination], destination_root)
+
+        # Thor::Error is the idiomatic way to abort a Rails generator. Thor
+        # catches it, prints the message in red, and exits with status 1
+        # without a Ruby backtrace. exit 1 would skip the framework's
+        # cleanup hooks and is hard to test.
+        raise Thor::Error, "Source file not found: #{options[:source]}" unless File.exist?(source_path)
+
+        ruby_source = ConfigConverter.from_yaml(source_path)
+        create_file destination_path, ruby_source
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply lib/generators/pgbus/update_generator.rb]: Fixed in eb5698f. Extracted a private `load_and_convert` helper that rescues `ConfigConverter::Error`, `Psych::Exception`, `Errno::ENOENT`, and `Errno::EACCES` and re-raises them as `Thor::Error` so failures surface with the standard generator behavior (red error, exit 1, no backtrace).
